packages/flutter_tools/lib/src/doctor.dart

```diff
@@ -594,6 +597,38 @@ class FlutterValidator extends DoctorValidator {
     return ValidationMessage.hint(flutterVersionMessage);
   }
 
+  List<ValidationMessage> _validateRequiredBinaries(String flutterRoot) {
+    final ValidationMessage? flutterWarning = _validateSdkBinary('flutter', flutterRoot);
+    final ValidationMessage? dartWarning = _validateSdkBinary('dart', flutterRoot);
+    return <ValidationMessage>[
+      if (flutterWarning != null) flutterWarning,
+      if (dartWarning != null) dartWarning,
+    ];
+  }
+
+  /// Return a warning if the provided [binary] on the user's path does not
+  /// resolve within the Flutter SDK.
+  ValidationMessage? _validateSdkBinary(String binary, String flutterRoot) {
+    final String flutterBinDir = _fileSystem.path.join(flutterRoot, 'bin');
+
+    final File? flutterBin = _operatingSystemUtils.which(binary);
+    if (flutterBin == null) {
```

Comment:
Should an extra info message be added if `flutterBinDir` points to a  different `flutterBin` eg: if a user have 2 installation of flutter and runs `/extra_install/flutter/bin/flutter doctor` and `flutter` points to `/first_install/flutter/bin/flutter`?

Reply:
I believe the check on line 622 should cover that?